File: Euler OLD/pal.java
public class pal{
	public static boolean palHelp(int n, int temp){
		if(n>=0 && n<10)
			return (n==temp%10);
		if(!palHelp(n/10, temp))
			return false;
		temp /= 10;
		return (n%10==temp%10);
	}

	public static boolean isPal(int n){
		int tmp = n;
		return palHelp(n,tmp);
	}

	public static void main(String[] args){
		int test = 1001;
		System.out.println(test + ": " + isPal(test));
	}
}

File: Euler OLD/pal.c
#include <stdio.h>
#include <math.h>

int isPal(int n, int* dup){
	if( n>=0 && n<10 )//1dig
		return (n==(*dup)%10);
	if(!isPal(n/10, dup)){
		return 0;
	}
	*dup/=10;
	return (n%10==(*dup)%10);
}

int palindrome(int n){
	int *d = &n;
	return isPal(n, d);
}

int main(){
	int test = 1000;

	printf("%i: %i.", test, palindrome(test));

	return 0;
}